File: src/JavaScript/11_Napisy_wprowadzenie/Zad18.js
var Main = (function() {
    function Main() {}
    Main.odwroc = function(slowo) {
        var rozdziel = slowo.split("");
        var odwroc = rozdziel.reverse();
        return odwroc.join("");
    };
    Main.odwrocNapisyV1 = function(napis) {
        var slowa = napis.split(" ");

        for (var i = 0; i < slowa.length; i++) {
            slowa[i] = Main.odwroc(slowa[i]);
        }

        return slowa.join(" ").trim();
    };
    Main.test1 = function() {
        var napis = "Cameron Diaz";
        var oczekiwane = "noremaC zaiD";
        var wynik = Main.odwrocNapisyV1(napis);
        if (!(wynik === oczekiwane)) {
            throw new Error(`Assertion error line 24: ${wynik} === ${oczekiwane}`);
        }
    };
    Main.main = function(args) {
        Main.test1();
    };
    return Main;
}());
Main["__class"] = "Main";
Main.main(null);

File: src/Python/11_Napisy_wprowadzenie/Zad18.py
import string

"""
Otrzymujesz napis reprezentujący zdanie. Zamień słowa zdania, na te 
same napisy zapisane wspak. Znaki interpunkcyjne nie są liczone jako słowa. 
"""


def podziel_zdanie_na_slowa(zdanie):
    return zdanie.translate(str.maketrans("", "", string.punctuation)).split()


def odwroc(napis):
    return napis[::-1]


def odwroc_slowa(zdanie):

    wynik = " ".join([odwroc(slowo) for slowo in podziel_zdanie_na_slowa(zdanie)])

    for i, znak in enumerate(zdanie):
        if znak in string.punctuation:
            wynik = wynik[:i] + znak + wynik[i:]

    return wynik


def test_odwroc_slowa():
    assert odwroc_slowa("Ala ma kota.") == "alA am atok."


if __name__ == "__main__":

    test_odwroc_slowa()
